Tests/RegexTests/MatchTests.swift

```diff
@@ -1575,27 +1577,57 @@ extension RegexTests {
       (input: "hezllo", match: nil),
       (input: "helloz", match: nil))
 
+    // MARK: Lookbehinds
     firstMatchTest(
-      #"(?<=USD)\d+"#, input: "Price: USD100", match: "100", xfail: true)
+      #"(?<=USD)\d+"#, input: "Price: USD100", match: "100")
     firstMatchTest(
-      #"(*plb:USD)\d+"#, input: "Price: USD100", match: "100", xfail: true)
+      #"(*plb:USD)\d+"#, input: "Price: USD100", match: "100")
     firstMatchTest(
       #"(*positive_lookbehind:USD)\d+"#,
-      input: "Price: USD100", match: "100", xfail: true)
-    // engines generally enforce that lookbehinds are fixed width
+      input: "Price: USD100", match: "100")
+
+    // TODO: Why is a match not found when unoptimized?
     firstMatchTest(
-      #"\d{3}(?<=USD\d{3})"#, input: "Price: USD100", match: "100", xfail: true)
+      #"\d{3}(?<=USD\d{3})"#, input: "Price: USD100", match: "100", validateOptimizations: false)
```

Comment:
In the unoptimized case we emit `reverse(1)`, I wonder if that's related?

Reply:
I ended up fixing this test case by checking if we're at the `start` during `reverseMatch` _and_ `reverseScalarMatch`